src/content/ui/adaptive-responsive/best-practices.md

```diff
@@ -0,0 +1,218 @@
+  Learn the best practices for creating >
+  a responsive and an adaptive app.
+short-title: Best practices
+---
+
+The best practices on this page cover,
+in particular, large screen devices,
+but can equally apply to any Flutter app.
+Flutter defines [large screens][] as tablets,
+foldables, and ChromeOS devices running Android.
+
+:::secondary Why do large screens matter, in particular?
+Demand for large screens continues to increase.
+As of January 2024,
+more than [270 million active large screen][large screens]
+and foldable devices run on Android.
+When your app supports large screens,
+it also receives the following benefits:
+
+* Optimizing your app to fill the screen improves
+  your app's user engagement metrics.
+* Increases your app's visibility in the Play Store.
+  Recent [Play Store updates][] show ratings by
+  device type and indicates when an app lacks
+  large screen support. 
+:::
+
+[large screens]: {{site.android-dev}}/guide/topics/large-screens/get-started-with-large-screens
+[Play Store updates]: {{site.android-dev}}/2022/03/helping-users-discover-quality-apps-on.html
+
+These best practices cover responsive _and_
+adaptive app design.
+
+## SafeArea
+
+When running your app on the latest devices,
+you might encounter bits of the UI being blocked
+by cutouts on top of the device's screen.
+
+You can fix this with the [`SafeArea`][] widget,
+which insets its child widget to avoid intrusions
+(like notches and camera cutouts),
+as well as operating system UI (such as the status bar
+on Android, or by rounded corners of the display).
+The `SafeArea` widget allows you to enable or
+disable padding on any of its four sides.
+
+It’s generally recommended to wrap the body of a
+`Scaffold` widget in `SafeArea` as a safe place to start,
+but you don’t always need to put it this high in the
+`Widget` tree.
+
+<b>TODO(sfshaza): PENDING: example code</b>
+
+For example, if you purposefully want your app to stretch
+under the cutouts, you can move the `SafeArea` to wrap
+whatever content makes sense,
+and let the rest of the app take up the full screen.
+
+<b>TODO(sfshaza): PENDING: example</b>
+
+Using `SafeArea` ensures that your app content won’t be
+cut off by physical display features or operating system UI
+and sets your app up for success even as new devices with
+different shapes and styles of cutouts enter the market.
+
+[`SafeArea`]: {{site.api}}/flutter/widgets/SafeArea-class.html
+
+## Don't gobble up all horizontal space
+
+If viewing your mobile app on a large screen device,
+you might notice that either a box or some text might
+take up all the horizontal space.
+
+This is _not_ recommended by the Android Large Screen App
+Quality Guidelines.
+There are a couple ways you can solve this.
+
+You can use the `GridView` widget to transform your
+existing `ListView` into more reasonably sized items,
+or you can use the `maxWidth` property of `BoxConstraints`
+to not have the list go edge-to-edge.
+
+### GridView
```

Comment:
### Avoid checking for types of hardware to make layout decisions. 

Avoid using Platfrom.isAndroid/isIOS directly or writing your own isMobile function. 

What space your app is actually given to render isn’t always tied to the full screen size of the device. Flutter can run on many different platforms, and your app may be running in a resizeable window on ChromeOS, side by side with another app on tablets in a multi-window mode, or even in a picture-in-picture on phones. Therefore, device type and app window size aren’t really strongly connected.

Instead, use MediaQuery to get the size of the window your app is currently running in. If you are tailoring the experience your user check out the section on `Policy and Capability`() or [watch this talk on youtube.](https://www.youtube.com/watch?v=GAnSNplNpCA&ab_channel=Flutter)

TODO I need to write you a section on Policy and Capability.